
From: NeilBrown <neilb@cse.unsw.edu.au>

When we upgrade an open, we keep the same stateid and bump only the seqid.

Signed-off-by: Andy Adamson <andros@umich.edu>
Signed-off-by: J. Bruce Fields <bfields@citi.umich.edu>
Signed-off-by: Neil Brown <neilb@cse.unsw.edu.au>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/nfsd/nfs4state.c |    2 --
 1 files changed, 2 deletions(-)

diff -puN fs/nfsd/nfs4state.c~nfsd-remove-incorrect-stateid-modification-in-nfsv4-open-upgrade fs/nfsd/nfs4state.c
--- 25/fs/nfsd/nfs4state.c~nfsd-remove-incorrect-stateid-modification-in-nfsv4-open-upgrade	2004-09-23 22:13:20.880064664 -0700
+++ 25-akpm/fs/nfsd/nfs4state.c	2004-09-23 22:13:20.885063904 -0700
@@ -1333,8 +1333,6 @@ nfsd4_process_open2(struct svc_rqst *rqs
 		/* remember the open */
 		set_bit(open->op_share_access, &stp->st_access_bmap);
 		set_bit(open->op_share_deny, &stp->st_deny_bmap);
-		/* bump the stateid */
-		update_stateid(&stp->st_stateid);
 	}
 	dprintk("nfs4_process_open2: stateid=(%08x/%08x/%08x/%08x)\n\n",
 	            stp->st_stateid.si_boot, stp->st_stateid.si_stateownerid,
_
